internal/querynodev2/delegator/delegator.go

```diff
@@ -172,6 +174,23 @@ func (sd *shardDelegator) SyncDistribution(ctx context.Context, entries ...Segme
 	sd.distribution.AddDistributions(entries...)
 }
 
+// SyncDistribution revises distribution.
+func (sd *shardDelegator) SyncPartitionStats(ctx context.Context, partVersions map[int64]int64) {
+	log := sd.getLogger(ctx)
+	log.Info("update partition stats versions")
+	sd.loadPartitionStats(ctx, partVersions)
+}
+
```

Comment:
Rlock protect get GetPartitionStatsVersions.
WLock protection is inside loadPartitionStats(ctx, partVersions)